torch/_higher_order_ops/triton_kernel_wrap.py

```diff
@@ -1068,6 +1067,21 @@ def init_variable(
                         "use_cuda_graph" in defaults
                         and defaults["use_cuda_graph"].default != kernel.use_cuda_graph
                     )
+                    # pre_hook requires running arbitrary code at runtime, which we cannot handle at this time
+                    # https://github.com/pytorch/pytorch/issues/139059
+                    or (
+                        # Check Config passed to autotuner in configs
+                        any(cfg.pre_hook is not None for cfg in kernel.configs)
+                    )
+                    # we also cannot support pre/post hook in the autotuner config
+                    or (
+                        "pre_hook" in defaults
+                        and defaults["pre_hook"] != kernel.pre_hook
+                    )
+                    or (
+                        "post_hook" in defaults
+                        and defaults["post_hook"] != kernel.post_hook
+                    )
```

Comment:
Seems the `pre_hook` and `post_hook` attributes of `Autotuner` are lambdas by default: https://github.com/triton-lang/triton/blob/f484cb8f00e4222c7db81b47c58e88467fe7a56b/python/triton/runtime/autotuner.py#L55-L56 . So these conditions will always be `True`. I don't see an obvious way to check if the `pre_hook` / `post_hook` have been passed (and are not defaults), as we don't intercept the `@triton.autotune` call in Dynamo, just the object that is used in the PT2-compiled code.

Perhaps, let's limit this PR to configs now and add flags to upstream Triton to mark that the `Autotuner`'s pre_/post_hook have been set? Then we can look at those flags here to error out.